assets/scss/modules/_carousel.scss

```
@@ -2,9 +2,193 @@
+      display: none;
+      min-height: 205px;
+      padding: $dp-spaces--lv3;
+      animation: fade-slide 1s;
+
+      @keyframes fade-slide {
+        0% {
+          opacity: 0;
+        }
+        100% {
+          opacity: 1;
+        }
+      }
+
+      @include media-breakpoint('mobile') {
+        min-height: 300px;
+      }
+
+      &:first-child {
+        display: block;
+      }
+
+      span {
+        font-size: 15px;
+        font-weight: 700;
+        color: $dp-color-lightgrey;
+        margin-bottom: $dp-spaces--lv3;
+        display: block;
+      }
+
+      h3 {
+        font-size: 23px;
+        font-weight: 400;
+        letter-spacing: 0.5px;
+        color: $dp-color-darkgrey;
+        padding-bottom: $dp-spaces--lv5;
+      }
+
+      a {
+        position: absolute;
+        bottom: 12px;
+        right: 20px;
+        padding-right: $dp-spaces--lv4;
+        display: inherit;
+        font-size: 15px;
+        font-weight: 700;
+        text-decoration: none;
+        color: $dp-color-lightgrey;
+        cursor: pointer;
+
+        span {
+          content: '';
+          display: block;
+          margin: 0;
+          position: absolute;
+          right: 0;
+          bottom: 0;
+          width: 18px;
+          height: 18px;
+          border-radius: 100%;
+          background: $dp-color-lightgrey;
+          text-align: center;
+          font-size: 8px;
+          line-height: 18px;
+          color: $dp-color-white;
+        }
+      }
+    }
+
+    &-dots {
+      position: absolute;
+      bottom: 0;
+      left: 0;
+      width: 100%;
+      display: inline-flex;
+      justify-content: left;
+
+      .dp-carousel-dot {
+        position: relative;
+        width: 14px;
+        height: 14px;
+        box-shadow: none;
+        margin-right: $dp-spaces--lv2;
+        cursor: pointer;
+      }
+
+      .dp-carousel-dot::after {
+        content: '';
+        position: absolute;
+        top: -1px;
+        left: -1px;
+        width: 16px;
+        height: 16px;
+        border-radius: 100%;
+        background: $dp-color-snow;
+        opacity: 1;
+        transition-duration: 0.2s;
+      }
+
+      .dp-carousel-dot:hover:after {
+        background: $dp-color-cement;
+        transition-duration: 0.2s;
+      }
+
+      .dp-carousel-dot:checked:after {
+        background: $dp-color-cement;
+      }
+
+      .dp-carousel-dot:focus:after {
+        box-shadow: inset 0 0 0 2px $dp-color-orange-focus;
+      }
+    }
+
+    &-orange {
+      border: 2px solid $dp-color-orange-carousel-border;
+      background: $dp-color-orange-carousel-back;
+
+      .dp-carousel-content .dp-carousel-slide span {
+        color: $dp-color-orange-carousel;
+      }
+
+      .dp-carousel-content .dp-carousel-slide a {
+        color: $dp-color-orange-carousel;
+      }
+
+      .dp-carousel-content .dp-carousel-slide a span {
+        background: $dp-color-orange-carousel;
+        color: $dp-color-white;
+      }
```

This could be a mixin right @GusBaamonde ?

---

@ornella-688  this example may help you, you can extend a class and use inheritance that is simpler and semantic:

.boton {
  font-size: 24px;
  padding: .5em .8em;
  border-radius: 3px;
  border: none;
  cursor: pointer;
}
.boton-principal {
  @extend .boton;
  background: #0BE66A;
}
.boton-secundario {
  @extend .boton;
  background: #DDDDDD;
}

---

I've tried a mixin like this:
```
@mixin colorSlide($primary-color, $border-color, $back-color) {
  border: 2px solid $border-color;
  background: $back-color;

  .dp-carousel-content .dp-carousel-slide span,
  .dp-carousel-content .dp-carousel-slide a {
    color: $primary-color;
  }

  .dp-carousel-content .dp-carousel-slide a span {
    background: $primary-color;
    color: $dp-color-white;
  }
}
```
Then here you should've replace the class content like this
```suggestion
    &-orange {
     @include colorSlide($dp-color-orange-carousel, $dp-color-orange-carousel-border, $dp-color-orange-carousel-back);
```
What do you think @GusBaamonde @ornella-688 ?

---

Ok, I see it

---

Don`t use the mixin, I've tried here https://www.sassmeister.com/ and it does add a lot of redundant css 😓 Thanks @GusBaamonde for this tool! It is a better approach to use inheritance.

---

I think that in this case I can't use inheritance because the styles inside the `orange` and `purple` classes are not the same. It could be in the case that they were the same.

---

I could only decrease the code like this:

```
   .dp-carousel-orange {
      border-color: $dp-color-orange-carousel-border;
      background: $dp-color-orange-carousel-back;

      .dp-carousel-content .dp-carousel-slide p,
      .dp-carousel-content .dp-carousel-slide a {
        color: $dp-color-orange-carousel;
      }

      .dp-carousel-content .dp-carousel-slide a span {
        background: $dp-color-orange-carousel;
      }
    }
```